tests/components/pushbullet/test_notify.py

```diff
@@ -1,301 +1,273 @@
+from tests.common import assert_setup_component, load_fixture
 
 
-class TestPushBullet(unittest.TestCase):
-    """Tests the Pushbullet Component."""
-
-    def setUp(self):
-        """Initialize values for this test case class."""
-        self.hass = get_test_home_assistant()
-        self.addCleanup(self.tear_down_cleanup)
-
-    def tear_down_cleanup(self):
-        """Stop everything that we started."""
-        self.hass.stop()
-
-    @patch.object(
+@pytest.fixture
+def mock_pushbullet():
+    """Mock pushbullet."""
+    with patch.object(
         PushBullet,
         "_get_data",
         return_value=json.loads(load_fixture("pushbullet_devices.json")),
-    )
-    def test_pushbullet_config(self, mock__get_data):
-        """Test setup."""
-        config = {
-            notify.DOMAIN: {
-                "name": "test",
-                "platform": "pushbullet",
-                "api_key": "MYFAKEKEY",
-            }
-        }
-        with assert_setup_component(1) as handle_config:
-            assert setup_component(self.hass, notify.DOMAIN, config)
-        assert handle_config[notify.DOMAIN]
+    ):
+        yield
 
-    def test_pushbullet_config_bad(self):
-        """Test set up the platform with bad/missing configuration."""
-        config = {notify.DOMAIN: {"platform": "pushbullet"}}
-        with assert_setup_component(0) as handle_config:
-            assert setup_component(self.hass, notify.DOMAIN, config)
-        assert not handle_config[notify.DOMAIN]
 
-    @requests_mock.Mocker()
-    @patch.object(
-        PushBullet,
-        "_get_data",
-        return_value=json.loads(load_fixture("pushbullet_devices.json")),
-    )
-    def test_pushbullet_push_default(self, mock, mock__get_data):
-        """Test pushbullet push to default target."""
-        config = {
-            notify.DOMAIN: {
-                "name": "test",
-                "platform": "pushbullet",
-                "api_key": "MYFAKEKEY",
-            }
+async def test_pushbullet_config(hass, mock_pushbullet):
+    """Test setup."""
+    config = {
+        notify.DOMAIN: {
+            "name": "test",
+            "platform": "pushbullet",
+            "api_key": "MYFAKEKEY",
         }
-        with assert_setup_component(1) as handle_config:
-            assert setup_component(self.hass, notify.DOMAIN, config)
-        assert handle_config[notify.DOMAIN]
-        mock.register_uri(
-            requests_mock.POST,
-            "https://api.pushbullet.com/v2/pushes",
-            status_code=200,
-            json={"mock_response": "Ok"},
-        )
-        data = {"title": "Test Title", "message": "Test Message"}
-        self.hass.services.call(notify.DOMAIN, "test", data)
-        self.hass.block_till_done()
-        assert mock.called
-        assert mock.call_count == 1
+    }
+    with assert_setup_component(1) as handle_config:
+        assert await async_setup_component(hass, notify.DOMAIN, config)
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6285789)